unimod_mapper/unimod_mapper.py

```diff
@@ -688,6 +719,165 @@ def _reparseXML(self, xml_file_list=[]):
+                mass = self.composition2mass(composition_unimod_style)
+                for i, name in enumerate(unimod_name_list):
+                    if name == unimod_name:
+                        unimod_id = unimod_id_list[i]
+                        unimod = True
+                        break
+                if unimod is False and unimod_name_list != []:
+                    logger.warning(
+                        "'{0}' is not a Unimod modification but the chemical composition you specified is included in Unimod. Please use one of the Unimod names: {1} Continue without modification {2} ".format(
+                            unimod_name, unimod_name_list, mod
+                        )
+                    )
+                    continue
+                if unimod is False and unimod_name_list == []:
+                    logger.warning(
+                        "'{0}' is not a Unimod modification trying to continue with the chemical composition you specified. This is not working with OMSSA so far".format(
+                            mod,
+                        )
+                    )
+                    mass = chemical_composition.mass()
+                    # write new userdefined modifications Xml in unimod style
+
+            mod_dict = copy.deepcopy(mod)
+            mod_dict.pop("type")
+            mod_dict.update({
+                "_id": index,
+                "aa": mod["aa"],
+                "mass": mass,
+                "pos": mod["position"],
+                "name": unimod_name,
+                "composition": composition,
+                "org": mod,
+                "id": unimod_id,
+                "unimod": unimod,
+                # "neutral_loss": neutral_loss
+            })
+
+            neutral_loss = []
+            # neutral_loss = [mod["neutral_loss"]]
+            if mod.get("neutral_loss", None) is not None:
+                if mod["neutral_loss"] == "unimod":
+                    # nl = self.name2nl_list(unimod_name)
+                    for nl_list in self.name2nl_list(unimod_name):
+                        for nl_item in nl_list:
+                            if nl_item[0] == mod_dict["aa"]:
+                                neutral_loss.append(nl_item[1])
+                else:
+                    neutral_loss.append(mod_dict["neutral_loss"])
+
+                mod_dict.update({"neutral_loss": neutral_loss})
+
+
+            # refactor the dict such as the first element of the list will be taken.
+            # Raise a warning if list has more than 1 entry!
+            # The double check allows to only modify those lists that were generated
+            # by the x2x_list functions, but accept lists in other mod_params
+            for obj in ["mass", "composition", "name", "id", "neutral_loss"]:
+                if obj in list(mod_dict.keys()):
+                    if isinstance(mod_dict[obj], list):
+                        if len(mod_dict[obj]) >= 1:
```

Comment:
That should be if `len(mod_dict[obj]) == 1` , right? Otherwise, it will never go into the elif!?

Reply:
yeah... thx for spotting >.> :)